model_compression_toolkit/core/keras/statistics_correction/apply_second_moment_correction.py

```diff
@@ -48,8 +48,13 @@ def keras_apply_second_moment_correction(quantized_model: Any,
                     .weights_second_moment_correction:
                 layer.trainable = True
 
-    for _ in tqdm(range(core_config.quantization_config.weights_second_moment_iters)):
-        quantized_model(representative_data_gen(), training=True)
+    i_iter = 0
+    while core_config.quantization_config.weights_second_moment_iters > i_iter:
+        for data in representative_data_gen():
```

Comment:
I know we talked about it offline but I still have a problem with this loops (also appears in GPTQ and maybe in other places):
First, the while loop with the inner "counter" that is increased inside an inner for loop + stops with a "break" is very "not pythonic" and very confusing. It calls for using a simple for loop over the number of iterations that we want, but the number of iterations can be altered by something that happens in the inner loop - I think it's an opening for bugs.
Second, we are basically wrapping the data sampler with a generator, just to peel off the generator and run over it like it is a simple list, and it adds this complication in the code (7 lines with 2 loops, and another loop inside the generator itself, instead of the 2 lines that were here before).
I don't see the point of that.